demos/sample_azure_iot_adu/sample_azure_iot_adu.c

```diff
@@ -495,8 +503,28 @@ static AzureIoTResult_t prvDownloadUpdateImageIntoFlash()
 
     LogInfo( ( "[ADU] Send HTTP request." ) );
 
+    ullPreviousTimeout = ullGetUnixTime();
+
     while( xImage.ulCurrentOffset < xImage.ulImageFileSize )
     {
+        ullCurrentTime = ullGetUnixTime();
+
+        if( ullCurrentTime - ullPreviousTimeout > ullTimeoutInSec )
+        {
+            LogInfo( ( "%i second timeout. Taking a break from downloading image.", ullTimeoutInSec ) );
+            LogInfo( ( "Receiving messages from IoT Hub." ) );
+            xResult = AzureIoTHubClient_ProcessLoop( &xAzureIoTHubClient,
```

Comment:
This is the easiest way to do this, but would it be the only way?
Have you thought about completely leaving prvDownloadUpdateImageIntoFlash and having only the  AzureIoTHubClient_ProcessLoop call in the main task?

Reply:
Yea I thought about that. I think it trades some simplicity for a different kind of complexity. Either you kick out and track the HTTP state outside the function, where resetting it also has to be done outside. Or you have multiple places to call the `_ProcessLoop()`. I can have another look and think about returning from this function instead.

Reply:
Yea looking at it again (and trying it a bit), it would add a different kind of complexity. I'm not totally opposed, but I only think it's worth it if people feel passionate about it.

Reply:
I think the decision is a function of time. If we are short on time, then this design should be enough. If we have more time, then the one you described would make this sample a little closer to a model customers could reuse.
I will approve this design and let you decide on what makes more sense. If you make any changes and need more review, please let me know.

Reply:
Understood. I think this works for the intended purpose, and with a subsequent update we can change if that's better 👍